server/offenceCodeDecisions/Decision.ts

```diff
@@ -0,0 +1,191 @@
+import Title from './Title'
+import Question from './Question'
+import Code from './Code'
+
+export default class Decision {
+  private parentDecision: Decision
+
+  private childrenDecisions: Array<Decision> = new Array<Decision>()
+
+  private readonly decisionQuestion: Question
+
+  private decisionCode: Code
+
+  private decisionTitle: Title
+
+  private decisionPage: Page
+
+  private decisionUrl: string
+
+  constructor(question?: Question | string) {
+    if (question instanceof Question) {
+      this.decisionQuestion = question
+    } else if (typeof question === 'string') {
+      this.decisionQuestion = new Question(question)
+    }
+  }
+
+  id(): string {
```

Comment:
We generate an id based on the index where the decision lives in the tree.
i.e. 0 is the root
0:0 is the roots first child etc.
As they are calculated like this the ids will change if we add or remove nodes in future - so we shouldn't persist them anywhere I think

Reply:
Yep - sounds ok. We only care about the end result, which is the `code`.